### DISCUSSION TITLE
keep aspect ratio for Picture-in-Picture

### DISCUSSION THREAD

**Discussion by Dolbaeb777K** [end-4/dots-hyprland#830] [Ideas]
We conferred in JaKooLit's discord and we figured out how to save the size of the `Picture-in-Picture`  and adjust its initial size, Look how it works:
 _I think you should to add it in your dots by default... But
`Picture-in-Picture` may be have another title because it depends on system language localization..._

https://github.com/user-attachments/assets/ce9c1e82-c6e9-4e52-bfe2-56adb9ee3398

When you turn on `Picture-in-Picture` mode it can be in absolutely any size, so you need to use the `size` variable in windowrulev2 but with only `X` argument, because otherwise black borders appear...

```
windowrulev2 = keepaspectratio, title:^(Picture-in-Picture)$

windowrulev2 = move 72% 7%,title:^(Picture-in-Picture)$ 
windowrulev2 = size 25%, title:^(Picture-in-Picture$

windowrulev2 = float, title:^(Picture-in-Picture)$
windowrulev2 = pin, title:^(Picture-in-Picture)$
```

**Comment by end-4** [comment]
alrighty, done in 08cd651fd14d885018e4729f014c0faf9902b358, thanks!
i also made it work with chrome too :) ~~weirdos who don't use a firefox or chromium based browser prob won't use my dotfiles so its not a problem~~ ~~but yes its still a problem if that title gets translated~~
